src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.Helpers.cs

```diff
@@ -113,6 +113,33 @@ private static void PopulateProperties(JsonTypeInfo typeInfo)
             }
         }
 
+        private const BindingFlags AllInstanceMembers =
+            BindingFlags.Instance |
+            BindingFlags.Public |
+            BindingFlags.NonPublic |
+            BindingFlags.DeclaredOnly;
+
+        /// <summary>
+        /// Looks up the type for a member matching the given name and member type.
+        /// </summary>
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        internal static MemberInfo? LookupMemberInfo(Type type, MemberTypes memberType, string name)
+        {
+            Debug.Assert(memberType is MemberTypes.Field or MemberTypes.Property);
+
+            // Walk the type hierarchy starting from the current type up to the base type(s)
+            foreach (Type t in type.GetSortedTypeHierarchy())
+            {
+                MemberInfo[] members = t.GetMember(name, memberType, AllInstanceMembers);
+                if (members.Length > 0)
+                {
+                    return members[0];
```

Comment:
What if there's more than one?

Reply:
Good point. I doubt this would create an issue here since indexers are excluded when properties are being populated (cf. method below this one).

Reply:
(I think it's also possible in IL for two properties to be named the same but have different return values.)